### DISCUSSION TITLE
Connect domain to yarp

### DISCUSSION THREAD

**Discussion by vynxc** [dotnet/yarp#2033] [Q&A] [answered]
Im trying to connect my server to my domain using yarp. This is how i current use nginx(i would like to swap to yarp) if somonencould convertert this that would be cool. I have tried and failed fornmany hours tonight
```
server {
    listen        80;
    server_name   example.com ;
    location / {
        proxy_pass         http://localhost:5000;
        
        
    }
}
```

**Comment by Tratcher** [comment]
Try this:
```json
{
  // Base URLs the server listens on, must be configured independently of the routes below
  "Urls": "http://example.com:80;https://example.com:443",
  "ReverseProxy": {
    "Routes": {
      "route1" : {
        "ClusterId": "cluster1",
        "Match": {
          "Path": "{**catch-all}",
        },
      }
    },
    "Clusters": {
      "cluster1": {
        "Destinations": {
          "cluster1/destination1": {
            "Address": "http://localhost:5000/"
          }
        }
      }
    }
  }
}

**Reply by vynxc** [reply]
And where should i configure ssl? kestrel with .pem cer and key doesnt seem to work for me.

**Reply by Tratcher** [reply]
https://learn.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/endpoints?view=aspnetcore-7.0#replace-the-default-certificate-from-configuration

**Reply by vynxc** [reply]
> https://learn.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/endpoints?view=aspnetcore-7.0#replace-the-default-certificate-from-configuration

i mean ssl on the domain

**Reply by Tratcher** [reply]
Can you expand on that?

**Reply by karelz** [reply]
@chaycee ping to @Tratcher's question?

**Reply by vynxc** [reply]
> Can you expand on that?

basically i am trying to replace nginx completely, but it seems i cannot yarp as a reverse proxy and have ssl, infact i couldnt even get the reverse proxy to work at all without ssl.

**Reply by Tratcher** [reply]
Unfortunately your questions aren't specific enough for us to help you. We'd need to see how you configured it, what errors you got, etc..

**Reply by baruba-cons** [reply]
It's worth noting that `Hosts` as described in [Configuration Structure](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/servers/yarp/config-files#configuration-structure) can also be used in the route matching process.

```json
"Match": {
  "Path": "{**catch-all}",
  "Hosts" : [ "www.aaaaa.com", "www.bbbbb.com"]
}
```

[Host matching in routes](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/routing#host-matching-in-routes-with-requirehost)

**Comment by karelz** [comment, marked as answer]
Triage: We should have proper migration docs from NGinx to YARP. They are useful.
Filed tracking issue #2037
